core/domain/user_jobs.py

```diff
@@ -233,6 +269,40 @@ def map(item):
                                 'type': 'exploration',
                                 'id': item.id
                             })
+        elif isinstance(item, collection_models.CollectionRightsModel):
+            if item.deleted:
```

Comment:
Because 'yield' cannot happen in a separate private method (at least I couldn't get it to work). Since most of the code is 'yield' it seemed like most of the duplication wouldn't be removed or I would have to introduce some ugly patterns (such as returning objects which are yielded). I don't like the duplication much, but it's cleaner and easier to maintain here.